diff --git a/c-stack.c b/c-stack.c
index 5353c08660d3e8d00e11a53f8af04e8a1f317a02..b778c0cd8f3a5c822d9448deb32cbfb60bb546bb 100644
--- a/c-stack.c
+++ b/c-stack.c
@@ -52,13 +52,15 @@ typedef struct sigaltstack stack_t;
 #endif
 #ifndef SIGSTKSZ
 # define SIGSTKSZ 16384
-#elif HAVE_LIBSIGSEGV && SIGSTKSZ < 16384
+#elif HAVE_LIBSIGSEGV
+# if defined(SIGSTKSZ) && (SIGSTKSZ) < 16384
 /* libsigsegv 2.6 through 2.8 have a bug where some architectures use
    more than the Linux default of an 8k alternate stack when deciding
    if a fault was caused by stack overflow.  */
 # undef SIGSTKSZ
 # define SIGSTKSZ 16384
 #endif
+#endif
 
 #include <stdlib.h>
 #include <string.h>
