--- a/drivers/usb/serial/usb-serial.c
+++ b/drivers/usb/serial/usb-serial.c
@@ -755,6 +755,18 @@ static int usb_serial_probe(struct usb_interface *interface,
 		return -ENOMEM;
 	}
 
+    if ((le16_to_cpu(dev->descriptor.idVendor)== 0x19d2)&&
+
+(le16_to_cpu(dev->descriptor.idProduct)== 0x0199)){
+
+if(1 ==interface->cur_altsetting->desc.bInterfaceNumber){
+
+return -ENODEV;
+
+}
+
+}
+
 	/* if this device type has a probe function, call it */
 	if (type->probe) {
 		const struct usb_device_id *id;
